src/plugins/data/public/search/aggs/types.ts

```diff
@@ -70,3 +72,25 @@ export interface SearchAggsStart {
   ) => InstanceType<typeof AggConfigs>;
   types: AggTypesRegistryStart;
 }
+
+/** @internal */
+export interface AggExpressionType {
+  type: 'agg_type';
+  value: CreateAggConfigParams;
+}
```

Comment:
This is the shape of the `agg_type` which every agg type function returns. That way we can enforce that `esaggs` only accept `agg_type` as an argument, instead of a stringified JSON blob.